src/test/java/com/google/graphgeckos/dashboard/storage/DatastoreRepositoryTests.java

```diff
@@ -209,4 +218,33 @@ public void testGetLastRevisionsQueryAfterDeletion() throws IOException, Interru
     Assert.assertEquals(results.get(0), dummy2);
     Assert.assertEquals(results.get(1), dummy1);
   }
+
+  @Test
+  public void testValidGetAndSetIndexRequests() throws IOException, InterruptedException {
+    DatastoreRepository storage = new DatastoreRepository(emulator.getOptions().getService());
+
+    storage.setBuildbotIndex("tester", 1000);
+    
+    Assert.assertEquals(1000, storage.getBuildbotIndex("tester"));
+  }
+
+  @Test
+  public void testInvalidGetIndexRequest() throws IOException, InterruptedException {
+    DatastoreRepository storage = new DatastoreRepository(emulator.getOptions().getService());
+
+    Assert.assertThrows(NotBoundException.class, () -> {
+      storage.getBuildbotIndex("tester");
+    });
+  }
+
+  @Test
+  public void testInvalidSetIndexRequest() throws IOException, InterruptedException {
+    DatastoreRepository storage = new DatastoreRepository(emulator.getOptions().getService());
```

Comment:
Consider adding a line here that shows that equal index leads to an exception as well.

Reply:
Done, thank you!